src/datatypes/src/error.rs

```diff
@@ -139,6 +140,14 @@ pub enum Error {
         error: arrow::error::ArrowError,
         location: Location,
     },
+
+    #[snafu(display("Failed to convert Arrow array {:?} to scalars", array))]
+    ConvertArrowArrayToScalars {
+        array: ArrayRef,
```

Comment:
Printing the whole array in the error message maybe too verbose if the array is large. Hopes DataFusionError has enough information.